### ISSUE TITLE
Agent is disconnected and server rebooted before connection is restored

### ISSUE THREAD

**Issue by trocade** [wazuh/wazuh#11217]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.2.5 | Wazuh component | Agent | Packages | ubuntu 18.04.6 LTS |

**Scenario:**
Agent is disconnected due to network issue, and during the outtage the system is rebooted, network connectivity is restored after reboot.

**Excepted behavior:**
Logs generated during agent disconnect up-to reboot should be processed by the agent and sent to manager.

**Actual behavior:**
All events generated between disconnect and reboot are lost.

**How this was tested:**
```
iptables -A OUTPUT -p tcp --dport 1514 -j DROP
useradd -s /bin/bash -m suspicioususer
reboot
```

The scenario where connection is restored before reboot behaves as expected where events are sent from the agent to manager.
Client buffer is enabled in ossec.conf
```
  <client_buffer>
    <!-- Agent buffer options -->
    <disabled>no</disabled>
    <queue_size>5000</queue_size>
    <events_per_second>500</events_per_second>
  </client_buffer>

```

**Comment by juliancnn** [comment]
Hi @trocade,

The mechanism of the buffer (Anti-flooding mechanism) is designed to prevent large bursts of events on an agent from negatively impacting the network or the manager.  Its functionality is not to store events in a cache and flush them to disk.

You could read more about this buffer [here](https://documentation.wazuh.com/current/user-manual/capabilities/antiflooding.html).

If you don't mind, I will leave this issue marked as an improvement and open for our team to analyze it and decide if this should be implemented, estimate it and add it to our roadmap.

Please let me know if you found this helpful.

Regards,
Julian

**Comment by trocade** [comment]
Hi @juliancnn ,

It is understood. 
One idea of improvement could be that when connection is lost the agent stores the current file position(s)  and wait for connectivity to come back (maybe this is what it is doing already when it enters locked state) but from time to time also store this value to a file so it can continue from where it was after reboot.

**Comment by branchnetconsulting** [comment]
Yes, we need need logcollector to respond to queue backpressure.
See https://github.com/wazuh/wazuh/issues/11241

**Comment by vikman90** [comment]
Related issue: #8444.
